x-pack/plugins/observability_solution/apm/server/routes/entities/services/routes.ts

```diff
@@ -46,6 +58,99 @@ const servicesEntitiesRoute = createApmServerRoute({
   },
 });
 
+const servicesEntitiesDetailedStatisticsRoute = createApmServerRoute({
+  endpoint: 'POST /internal/apm/entities/services/detailed_statistics',
+  params: t.type({
+    query: t.intersection([
+      environmentRt,
+      kueryRt,
+      rangeRt,
+      t.intersection([offsetRt, probabilityRt, serviceTransactionDataSourceRt]),
+      t.type({
+        bucketSizeInSeconds: toNumberRt,
+      }),
+    ]),
+    body: t.type({ serviceNames: jsonRt.pipe(t.array(t.string)) }),
+  }),
+  options: { tags: ['access:apm'] },
+  handler: async (resources) => {
+    const {
+      context,
+      params,
+      request,
+      plugins: { security, logsDataAccess },
+    } = resources;
+
+    const [coreContext, logsDataAccessStart] = await Promise.all([
+      context.core,
+      logsDataAccess.start(),
+    ]);
+
+    const {
+      environment,
+      kuery,
+      offset,
+      start,
+      end,
+      probability,
+      documentType,
+      rollupInterval,
+      bucketSizeInSeconds,
+    } = params.query;
+
+    const { serviceNames } = params.body;
+
+    const [apmEventClient, randomSampler] = await Promise.all([
+      getApmEventClient(resources),
+      getRandomSampler({ security, request, probability }),
+    ]);
+
+    if (!serviceNames.length) {
```

Comment:
We could make this validation before the API calls to save time in case `serviceNames` is empty

Reply:
Good point, I missed that when I got this part from the service overview request, moved 👍